src/uct/ib/dc/dc_mlx5.c

```diff
@@ -882,11 +886,22 @@ static inline ucs_status_t uct_dc_mlx5_iface_flush_dcis(uct_dc_mlx5_iface_t *ifa
 ucs_status_t uct_dc_mlx5_iface_flush(uct_iface_h tl_iface, unsigned flags, uct_completion_t *comp)
 {
     uct_dc_mlx5_iface_t *iface = ucs_derived_of(tl_iface, uct_dc_mlx5_iface_t);
+    uct_ib_mlx5_md_t *md       = ucs_derived_of(iface->super.super.super.super.md,
+                                                uct_ib_mlx5_md_t);
+
     ucs_status_t status;
 
     if (comp != NULL) {
         return UCS_ERR_UNSUPPORTED;
     }
+
+    if (md->super.relaxed_order) {
+        status = uct_rc_iface_fence(tl_iface, 0);
```

Comment:
why do we need it for flush?

Reply:
To enforce order - otherwise operation executed before flush may be performed after operation executed after flush.

Reply:
shall we use `ops.iface_fence` instead to make sure we use correct fence if it gets changed for iface?

Reply:
also we need to do flush for `rc_x`, right?